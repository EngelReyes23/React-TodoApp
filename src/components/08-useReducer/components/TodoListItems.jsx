import React from "react";
import PropTypes from "prop-types";

const TodoListItems = ({ todo, index, handleDelete, handleToggle }) => {
	return (
		<li className="list-group-item" key={index}>
			<p
				className={`${todo.done && "complete"}`}
				onClick={() => handleToggle(todo.id)}
			>
				{++index}- {todo.description}
			</p>
			<button
				onClick={() => {
					handleDelete(todo.id);
				}}
				className="btn btn-danger"
			>
				Delete
			</button>
		</li>
	);
};
// si esta envuelto en un memo , en los test no se muestra como deberia ser
// TodoListItems.displayName = "TodoListItems";

// PropTypes requeridos
TodoListItems.propTypes = {
	todo: PropTypes.object.isRequired,
	index: PropTypes.number.isRequired,
	handleDelete: PropTypes.func.isRequired,
	handleToggle: PropTypes.func.isRequired,
};

export default React.memo(TodoListItems).displayName = "TodoListItems";
