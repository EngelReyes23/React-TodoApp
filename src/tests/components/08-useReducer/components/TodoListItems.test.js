import { shallow } from "enzyme";
import TodoListItems from "../../../../components/08-useReducer/components/TodoListItems";
import { demoTodo } from "../../../fixtures/demoTodo";

describe("Pruebas a <TodoListItems/>", () => {
	const handleDelete = jest.fn();
	const handleToggle = jest.fn();
	//
	// 1er Test - Testeo de renderizado
	test("debe de mostrar el componente correctamente", () => {
		const wrapper = shallow(
			<TodoListItems
				todo={demoTodo[1]}
				index={1}
				handleToggle={handleToggle}
				handleDelete={handleDelete}
			/>
		);
		expect(wrapper).toMatchSnapshot();
		// expect(wrapper.find("p").text().trim()).toBe(
		// 	`2- ${demoTodo[1].description}`
		// );
	});

	// 2do Test - Llamar la funcion handleDelete
	test("debe de llamar a la funcion handleDelete", () => {
		const wrapper = shallow(
			<TodoListItems
				todo={demoTodo[1]}
				index={1}
				handleToggle={handleToggle}
				handleDelete={handleDelete}
			/>
		);
		wrapper.find("button").simulate("click");
		expect(handleDelete).toHaveBeenCalledWith(demoTodo[1].id);
	});

	// 3er Test - Llamar la funcion handleToggle
	test("debe de llamar a la funcion handleToggle", () => {
		const wrapper = shallow(
			<TodoListItems
				todo={demoTodo[1]}
				index={1}
				handleToggle={handleToggle}
				handleDelete={handleDelete}
			/>
		);
		wrapper.find("p").simulate("click");
		expect(handleToggle).toHaveBeenCalledWith(demoTodo[1].id);
	});

	// 4to Test - Mostrar el texto del parrafo correctamente
	test("debe de mostrar el texto del parrafo correctamente", () => {
		const wrapper = shallow(
			<TodoListItems
				todo={demoTodo[1]}
				index={1}
				handleToggle={handleToggle}
				handleDelete={handleDelete}
			/>
		);
		expect(wrapper.find("p").text().trim()).toBe(
			`2- ${demoTodo[1].description}`
		);
	});

	// 5to Test - Debe tener la clase complete
	test("Debe tener la clase complete", () => {
		const newTodo = demoTodo[1];
		newTodo.done = true;

		const wrapper = shallow(
			<TodoListItems todo={newTodo} index={newTodo.id} />
		);
		expect(wrapper.find("p").hasClass("complete")).toBe(true);
	});
});
