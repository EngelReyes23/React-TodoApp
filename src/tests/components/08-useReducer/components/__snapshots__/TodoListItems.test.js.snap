// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <TodoListItems/> debe de mostrar el componente correctamente 1`] = `
<TodoListItems
  handleDelete={[MockFunction]}
  handleToggle={[MockFunction]}
  index={1}
  todo={
    Object {
      "description": "Engel Reyes",
      "done": false,
      "id": 2,
    }
  }
/>
`;
